### DISCUSSION TITLE
usbsdmux connection error

### DISCUSSION THREAD

**Discussion by AndreHochuliMT** [labgrid-project/labgrid#804] [General]
Hi,
I am setting up a new distributed integration test environment using labgrid. It a lot of fun, i able run the exporter on an segin-imx6ull board, the client and the coordinator an a debian laptop. 

**I can successfully run:**
labgrid-client -d -p example-place console
I am able to run usbsdmux /dev/sg0 get on exporter.

**I get an error running**
When executing labgrid-client -d -p example-place sd-mux get
[sdmuxClientError.txt](https://github.com/labgrid-project/labgrid/files/6855393/sdmuxClientError.txt)
[coordinatoroutput.txt](https://github.com/labgrid-project/labgrid/files/6855391/coordinatoroutput.txt)
[Exporteroutput.txt](https://github.com/labgrid-project/labgrid/files/6855392/Exporteroutput.txt)

**When running ssh command manually it looks good:**
```bash
ssh -x -o LogLevel=ERROR -o PasswordAuthentication=no -n -MN -o ConnectTimeout=30 -o ControlPersist=300 -o ControlMaster=yes -o ControlPath=/tmp/labgrid-connection-107699ek/control-segin-imx6ull-sdcd-2 -o StrictHostKeyChecking=yes segin-imx6ull-sdcd-2 -vvv
```
See attached file sshoutput

[sshoutput.txt](https://github.com/labgrid-project/labgrid/files/6855396/sshoutput.txt)

I have spent a few hours try to make it run, but i have reached a point where i am stuck.

It would be very nice if you could have a look through the files. I am sure that some experienced user can see my problem.

Kind regards André

**Comment by Emantor** [comment]
This really does look weird, especially since the SSH command succeeds outside of the labgrid-client call.
The failing call also doesn't contain any error, you can turn up the loglevel by modifying `-o LogLevel=ERROR` to something like `DEBUG` maybe this can shed some light on the problem.
The only other thing that comes to mind is to inspect the environments within the labgrid-client invocation and outside, maybe something like SSH agent forwarding lets it work without labgrid-client.

**Reply by AndreHochuliMT** [reply]
Thanks for the quick reply.

When setting the LogLevel to DEBUG i get the following output.
[sdmuxClientErrorLogLevelDebug.txt](https://github.com/labgrid-project/labgrid/files/6868210/sdmuxClientErrorLogLevelDebug.txt)

The directory /tmp/labgrid-connection-q4q4hqr_ is not created.
The user has enough rights to create folders, mkdir works in /tmp directory

I am new to pyhton, I was not able to find out why the /tmp/labgrid-connection-q4q4hqr_ folder is not created.

I will be in vacation until 9.Aug.

Thanks André

**Reply by Emantor** [reply]
So apparently the connection is created successfully but the exitcode is `2`.Are control sockets disabled on the remote side?

**Reply by jluebbe** [reply]
@AndreHochuliMT Is this problem still current?

**Reply by AndreHochuliMT** [reply]
I was not able to make it run on our own BSP. We are in the process of evaluating a integration test hardware and therefore I have stopped working on this issue.
